pkg/health/condition/check_backup_ready.go

```diff
@@ -0,0 +1,111 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package condition
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	coordinationv1 "k8s.io/api/coordination/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type backupReadyCheck struct {
+	cl client.Client
+}
+
+const (
+	BackupSucceeded string = "BackupSucceeded"
+	BackupFailed    string = "BackupFailed"
+	Unknown         string = "Unknown"
+)
+
+func (a *backupReadyCheck) Check(ctx context.Context, etcd druidv1alpha1.Etcd) Result {
+	//Default case
+	result := &result{
+		conType: druidv1alpha1.ConditionTypeBackupReady,
+		status:  druidv1alpha1.ConditionUnknown,
+		reason:  Unknown,
+		message: "Cannot determine etcd backup status since no snapshot leases present",
+	}
+
+	//Fetch snpashot leases
+	fullSnapLease := &coordinationv1.Lease{}
+	deltaSnapLease := &coordinationv1.Lease{}
+	var fullSnapErr, incrSnapErr error
+	fullSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getFullSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, fullSnapLease)
+	incrSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getDeltaSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, deltaSnapLease)
+
+	if fullSnapErr != nil || incrSnapErr != nil || (fullSnapLease.Spec.HolderIdentity == nil && deltaSnapLease.Spec.HolderIdentity == nil) {
+		return result
+	}
+
+	deltaLeaseRenewTime := deltaSnapLease.Spec.RenewTime
+	fullLeaseRenewTime := fullSnapLease.Spec.RenewTime
+	fullLeaseCreateTime := &fullSnapLease.ObjectMeta.CreationTimestamp
+
+	if fullSnapLease.Spec.HolderIdentity == nil && deltaLeaseRenewTime != nil && fullLeaseCreateTime != nil {
+		//Most probable during reconcile of existing clusters if fresh leases are created
+		//Treat backup as succeeded if delta snap lease renewal happens in the required time window and full snap lease not more than 24h old
+		if time.Since(deltaLeaseRenewTime.Time) < 2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration && time.Since(fullLeaseCreateTime.Time) < 24*time.Hour {
+			result.reason = BackupSucceeded
+			result.message = "Delta snapshot backup succeeded"
+			result.status = druidv1alpha1.ConditionTrue
+			return result
+		}
+	} else if deltaSnapLease.Spec.HolderIdentity == nil && fullLeaseRenewTime != nil {
+		//Most probable during a startup scenario for new clusters
+		//Special case. Return Unknown condition for some time to allow delta backups to start up
+		if time.Since(fullLeaseRenewTime.Time) < 5*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration {
```

Comment:
You're right, I see your point here

I still need we need this check, but I'll propose something slightly different on the whole
1. instead of having any `nil` checks for `HolderIdentity`, I'll instead use `RenewTime` as `HolderIdentity` won't be set without `RenewTime`. It should amount to the same thing, but will be more significant with point 2

2. I've already proposed a change in lease update logic in [etcd-backup-restore#410](https://github.com/gardener/etcd-backup-restore/pull/410) where even if a delta snapshot is not created, the snapshot lease is still renewed (needed as the lease renewals are heartbeats, and we shouldn't skip a beat if there's no change in the cluster)

With these two changes combined, I think the condition should make more sense as a whole

Reply:
> With these two changes combined, I think the condition should make more sense as a whole

Makes sense. However, we now have a dependency and can (strictly) only merge this PR when [etcd-backup-restore#410](https://github.com/gardener/etcd-backup-restore/pull/410) is merged and released. So either we set this PR on hold for the time being or we remove this check and add it again when etcd-br is released. I personally tend to do the latter so you don't have to worry about rebasing.

Reply:
makes sense
Removed this condition